lib/ClangImporter/ImportType.cpp

```diff
@@ -812,7 +812,7 @@ namespace {
         // Map anonymous enums with no fixed underlying type to Int /if/
         // they fit in an Int32. If not, this mapping isn't guaranteed to be
         // consistent for all platforms we care about.
-        if (!clangDecl->isFixed() &&
+        if (!clangDecl->isFixed() && clangDecl->isFreeStanding() &&
```

Comment:
I asked on the Clang list whether `isFreeStanding` or `isEmbeddedInDeclarator` is the appropriate query here; my initial impression is that they do the same thing.

Reply:
If they're the same shouldn't one of them be deleted?

Reply:
I'm assuming they're *not* the same and I'm just missing something, but if they *are* the same I bet someone will ask me to do that.

Reply:
Yes, I'm not sure why they wouldn't just be exact opposites.